### PR TITLE
Add flint/2.7.1

### LINKED ISSUE #3693: [request] flint/2.6.3
### Package Details
  * Package Name/Version: **flint/2.6.3**
  * Website: **http://www.flintlib.org/**
  * Source code: **http://www.flintlib.org/flint-2.6.3.tar.gz** (or on GitHub: **https://github.com/wbhart/flint2/archive/v2.6.3.tar.gz**)

### Description Of The Library / Tool

FLINT is a C library for doing number theory

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **flint/2.7.1**

Fixes #3693

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — DISMISSED** [review]

File: recipes/flint/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self.requires("mpfr/4.1.0")
+        if self.settings.compiler == "Visual Studio":
+            self.requires("pthreads4w/3.0.0")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```
Comment (madebr) [inline recipes/flint/all/conanfile.py:49]: Pleas add a `required_conan_version = ">=1.33.0"` at the top of this recipe.

File: recipes/flint/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
```
Comment (madebr) [inline recipes/flint/all/conanfile.py:52]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

File: recipes/flint/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (madebr) [inline recipes/flint/all/test_package/CMakeLists.txt:5]: `conan_basic_setup` without `TARGETS` will call `include_directories`
```suggestion
conan_basic_setup(TARGETS)
```

File: recipes/flint/all/conanfile.py (hunk 2 above)
Comment (theirix) [reply recipes/flint/all/conanfile.py:52]: Is it recommended for all new recipes, even with a single version?

File: recipes/flint/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (theirix) [reply recipes/flint/all/test_package/CMakeLists.txt:5]: nice catch, thanks!

File: recipes/flint/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/flint/all/conanfile.py]: There's no harm and it's future proof without code bloat.

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

File: recipes/flint/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        return self._cmake
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "libflint"
+        self.cpp_info.names["cmake_find_package_multi"] = "libflint"
```
Comment (SpaceIm) [inline recipes/flint/all/conanfile.py:81]: Ooops, completly missed this PR.

Where did you find the CMake Find/config file?

Comment (theirix) [reply recipes/flint/all/conanfile.py:81]: That's right, seems it is an ungrounded guess. Could you please file a followup fix?

**Pull request merged**